app/helpers/wifi_density_helper.rb

```diff
@@ -0,0 +1,94 @@
+module WifiDensityHelper
+  CACHE_DURATION = Rails.env.development? ? 10.seconds : 5.minutes
+
+  def wifi_density_percentage(location)
+    # If this location's code is not mapped in the wifi density config file,
+    # that means wifi density info is not supported for the location.
+    config_for_location = WIFI_DENSITY[location.code]
+    return nil if config_for_location.nil? || !config_for_location.key?('high') || !config_for_location.key?('cuit_location_ids')
+
+    Rails.cache.fetch('wifi_density_data-' + location.code, expires_in: CACHE_DURATION) do
+      wifi_density_data_for_locations = Rails.cache.fetch('wifi_density_data', expires_in: CACHE_DURATION) do
+        fetch_hierarchical_wifi_density_data
+      end
+      aggregated_locations = {}
+
+      config_for_location['cuit_location_ids'].each do |cuit_location_id|
+        location_data = wifi_density_data_for_locations[cuit_location_id.to_s]
+        next if location_data.nil?
+        aggregated_locations.merge!({cuit_location_id.to_s => location_data})
+        aggregated_locations.merge!(recursively_collect_children(location_data))
+      end
+
+      # Sum client_count totals from all aggregated locations
```

Comment:
I believe you should be able to use [Enumerable#reduce](https://ruby-doc.org/core-2.6/Enumerable.html#method-i-reduce) here.